skada/_reweight.py

```diff
@@ -669,3 +670,272 @@ def KLIEP(
+        Determines random number generation for dataset creation. Pass an int
+        for reproducible output across multiple function calls.
+
+    Attributes
+    ----------
+    `weights_` : array-like, shape (n_samples,)
+        The learned weights.
+    `X_source_` : array-like, shape (n_samples, n_features)
+        The source data.
+
+    References
+    ----------
+    .. [1] J. Huang, A. Gretton, K. Borgwardt, B. Schölkopf and A. J. Smola.
+           'Correcting sample selection bias by unlabeled data.'
+           In NIPS, 2007.
+    """
+
+    def __init__(
+        self,
+        kernel="rbf",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        B=1000.,
+        eps=None,
+        tol=1e-6,
+        max_iter=100,
+        random_state=None,
+    ):
+        super().__init__()
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.B = B
+        self.eps = eps
+        self.tol = tol
+        self.max_iter = max_iter
+        self.random_state = random_state
+
+        if kernel not in KERNEL_PARAMS:
+            raise ValueError("`kernel` argument should be included in %s,"
+                             " got '%s'" % (str(KERNEL_PARAMS), str(kernel)))
+
+    def fit(self, X, y=None, sample_domain=None, **kwargs):
+        """Fit adaptation parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, sample_domain = check_X_domain(
+            X,
+            sample_domain,
+            allow_multi_source=True,
+            allow_multi_target=True
+        )
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+
+        self.weights_ = self._weights_optimization(X_source, X_target)
+        self.X_source_ = X_source
+
+        return self
+
+    def _weights_optimization(self, X_source, X_target):
+        """Weight optimization"""
+        Kss = pairwise_kernels(X_source,
+                               metric=self.kernel,
+                               filter_params=True,
+                               gamma=self.gamma,
+                               degree=self.degree,
+                               coef0=self.coef0)
+        Kst = pairwise_kernels(X_source,
+                               X_target,
+                               metric=self.kernel,
+                               filter_params=True,
+                               gamma=self.gamma,
+                               degree=self.degree,
+                               coef0=self.coef0)
+        Ns = Kss.shape[0]
```

Comment:
Same comment as is did to @antoinecollas , we need a function for QP solver, we will use it many times probably. You should proably work together on this one with one merge before the other

Reply:
Also I'm sorry but you estimate a kernel machine reweighting sum_i w_i k(x,,x_i) my intuition is you should compute this rewieghting instead of juste the weights estimated on the points and then nearest neighbor (let us talk about this in slack)